src/test/java/org/elasticsearch/test/store/MockDirectoryHelper.java

```diff
@@ -122,11 +127,26 @@ public FsDirectoryService randomDirectorService(IndexStore indexStore) {
         private final boolean crash;
         private volatile RuntimeException closeException;
         private final Object lock = new Object();
+        private final Set<String> superUnSyncedFiles;
+        private final Random superRandomState;
 
         public ElasticsearchMockDirectoryWrapper(Random random, Directory delegate, ESLogger logger, boolean crash) {
             super(random, delegate);
             this.crash = crash;
             this.logger = logger;
+
+            // TODO: remove this silliness on Lucene upgrade:
+            try {
+                Field field = MockDirectoryWrapper.class.getDeclaredField("unSyncedFiles");
+                field.setAccessible(true);
+                superUnSyncedFiles = (Set<String>) field.get(this);
+
+                field = MockDirectoryWrapper.class.getDeclaredField("randomState");
+                field.setAccessible(true);
+                superRandomState = (Random) field.get(this);
```

Comment:
> Lucene's test-framework gets (justifiably) angry when it detects that you are sharing a single Random instance across threads

Hmm but that's not quite right either, because even after forking our private Random instance, it will be used from multiple threads that call MDW's APIs.  So now I'm less sure about how test-framework checks usage of Random across threads...